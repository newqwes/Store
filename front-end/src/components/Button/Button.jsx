import React from 'react';
import PropTypes from 'prop-types';

import THEME_VARIANT from '../../constants/themeVariant';

import ButtonWrapper from './styled';

const Button = ({ text, onClick, themeVariant }) => (
  <ButtonWrapper onClick={onClick} themeVariant={themeVariant}>
    {text}
  </ButtonWrapper>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  themeVariant: PropTypes.string,
};

Button.defaultProps = {
  themeVariant: THEME_VARIANT.default,
};

export default Button;
