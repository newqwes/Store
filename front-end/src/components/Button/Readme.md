Button example:

```jsx static
<Button text='Button' />
```

```jsx noeditor
<Button text='Button' />
```
