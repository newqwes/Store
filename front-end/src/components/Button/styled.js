import styled, { css } from 'styled-components';
import { get } from 'lodash/fp';

const ButtonWrapper = styled.button(
  ({ themeVariant }) => css`
    padding: 5px 8px;
    letter-spacing: 1px;
    background-color: ${get(['theme', 'button', 'background', themeVariant])};
    border: 1px solid ${get(['theme', 'button', 'border', themeVariant])};
    border-radius: 5px;

    color: ${get(['theme', 'button', 'color', themeVariant])};
    font-weight: 500;
    outline: none;
    font-size: 12px;
    white-space: nowrap;

    &:hover {
      cursor: pointer;
    }
  `,
);

export default ButtonWrapper;
