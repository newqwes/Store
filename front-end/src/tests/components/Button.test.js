import React from 'react';

import Button from '../../components/Button';
import { mountWithTheme } from '../helpers';

const initProps = {
  text: 'test text',
  onClick: () => {},
};

const setUp = (props) => mountWithTheme(<Button {...props} />);

describe('snapshot Button', () => {
  let component;

  beforeEach(() => {
    component = setUp(initProps);
  });

  it('snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('should render Button component', () => {
  let component;

  beforeEach(() => {
    component = setUp(initProps);
  });

  it('should contain text in button', () => {
    const button = component;

    expect(button.text()).toBe(initProps.text);
  });

  it('should contain only one button tag', () => {
    const wrapper = component.length;

    expect(wrapper).toBe(1);
  });
});
