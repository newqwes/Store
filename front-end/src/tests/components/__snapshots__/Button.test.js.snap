// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Button snapshot 1`] = `
<button
  className="sc-bdfBwQ cEQMOM"
  onClick={[Function]}
>
  test text
</button>
`;
